prov/efa/src/efa_hmem.c

```diff
@@ -142,6 +134,23 @@ static int efa_domain_hmem_info_init_cuda(struct efa_domain *efa_domain)
 	else
 		info->p2p_required_by_impl = true;
 
+	if (ofi_hmem_p2p_disabled()) {
+		info->p2p_disabled_by_user = true;
+		info->p2p_supported_by_device = false;
```

Comment:
Doesn't make sense to `p2p_supported_by_device` to false just because the user disables P2P. IOW, the user doesn't dictate the device's P2P capabilities; only whether or not they wish to utilize them.

Reply:
The reason I set these values for info is to do an early return when `ofi_hmem_p2p_disabled` so I can skip all the p2p checks, including ofi_cudaMalloc/neuron_alloc and ibv_reg_mr/ibv_reg_dmabuf_mr. 
Are you suggesting leaving `p2p_supported_by_device` unset?

Reply:
There's nothing wrong with an early exit when `ofi_hmem_p2p_disabled()` returns true. `p2p_supported_by_device` simply shouldn't depend on that condition. In the future, there may some other logic in the EFA provider that relies on `p2p_supported_by_device`; independent of the user's request to disable P2P HMEM operations.